src/menu_parser.py

```diff
@@ -241,65 +241,51 @@ def __get_price(canteen: Canteen, dish: Tuple[str, str, str, str, str], dish_nam
+        if page.ok:
+            try:
+                tree: html.Element = html.fromstring(page.content)
+                html_menus: List[html.Element] = self.get_daily_menus_as_html(tree)
+                for html_menu in html_menus:
+                    # this solves some weird reference? issue where tree.xpath will subsequently always use
+                    # the first element despite looping through seemingly separate elements
+                    html_menu = html.fromstring(html.tostring(html_menu))
+                    menu = self.get_menu(html_menu, canteen)
                     if menu:
-                        menus[date] = menu
+                        menus[menu.menu_date] = menu
                 # pylint: disable=broad-except
-                except Exception as e:
-                    print(f"Exception while parsing menu from {date}. Skipping current date. Exception args: {e.args}")
-                # pylint: enable=broad-except
+            except Exception as e:
+                print(f"Exception while parsing menu. Skipping current date. Exception args: {e.args}")
+        # pylint: enable=broad-except
         return menus
 
-    def get_menu(self, page: html.Element, canteen: Canteen, date: datetime.date) -> Optional[Menu]:
-        # get current menu
-        current_menu: html.Element = self.__get_daily_menus_as_html(page)[0]
-        # get html representation of menu
-        menu_html = html.fromstring(html.tostring(current_menu))
-
-        # parse dishes of current menu
-        dishes: List[Dish] = self.__parse_dishes(menu_html, canteen)
-        # create menu object
+    def get_menu(self, page: html.Element, canteen: Canteen) -> Optional[Menu]:
+        date = self.extract_date_from_html(page)
+        dishes: List[Dish] = self.__parse_dishes(page, canteen)
         menu: Menu = Menu(date, dishes)
         return menu
 
-    def __get_available_dates(self, canteen: Canteen) -> List[datetime.date]:
-        page_link: str = self.base_url.format(url_id=canteen.url_id)
-        page: requests.Response = requests.get(page_link, timeout=10.0)
-        tree: html.Element = html.fromstring(page.content)
-        return self.get_available_dates_for_html(tree)
-
     # public for testing
-    def get_available_dates_for_html(self, tree: html.Element) -> List[datetime.date]:
-        dates: List[datetime.date] = []
-        date_strings: List[str] = tree.xpath("//div[@class='c-schedule__item']//strong/text()")
-        for date_str in date_strings:
-            # parse date
-            try:
-                date: datetime.date = util.parse_date(date_str)
-            except ValueError:
-                print(f"Warning: Error during parsing date from html page. Problematic date: {date_str}")
-                # continue and parse subsequent menus
-                continue
-            dates += [date]
-        return dates
+    @staticmethod
+    def extract_date_from_html(tree: html.Element) -> Optional[datetime.date]:
```

Comment:
Python does not quite have a concept of public/private.
I can access private methods (those starting with `__`) without any problems from outside the module.
I might get a linting warning/.., but those can be ignored for tests

=> I would prefer the following and the comments can be removed. We can do this in a followup if you'd like

```suggestion
    def __extract_date_from_html(tree: html.Element) -> Optional[datetime.date]:
```